cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -496,18 +496,17 @@ impl ABIMachineSpec for X64ABIMachineSpec {
     }
 
     fn gen_clobber_save(
-        call_conv: isa::CallConv,
+        _call_conv: isa::CallConv,
+        setup_frame: bool,
         flags: &settings::Flags,
-        clobbers: &Set<Writable<RealReg>>,
+        clobbered_callee_saves: &Vec<Writable<RealReg>>,
         fixed_frame_storage_size: u32,
         _outgoing_args_size: u32,
     ) -> (u64, SmallVec<[Self::I; 16]>) {
         let mut insts = SmallVec::new();
-        // Find all clobbered registers that are callee-save.
-        let clobbered = get_callee_saves(&call_conv, clobbers);
-        let clobbered_size = compute_clobber_size(&clobbered);
+        let clobbered_size = compute_clobber_size(&clobbered_callee_saves);
 
-        if flags.unwind_info() {
+        if flags.unwind_info() && setup_frame {
```

Comment:
I can't say I have looked into this much but isn't there some way for the unwind info to still unwind leaf functions? Looking at the [test where MacOS fails](https://github.com/bytecodealliance/wasmtime/blob/e526ffc0d6bcc3431aee49a8037ef44478c4d112/tests/all/traps.rs#L140), I think this problem will also exist for aarch64?